### PR TITLE
[bug] Make 'is_sms_verification_enabled' respect global settings#301

### LINKED ISSUE #301: [bug] Make  "is_sms_verification_enabled" respect global settings
I have done some similar work in https://github.com/openwisp/openwisp-radius/pull/300 for `registration_enabled` which can be used as reference. 

https://github.com/openwisp/openwisp-radius/blob/602f3ecf357c3b4ecba0a60dabedd6f9c28f1319/openwisp_radius/base/models.py#L1142-L1145

https://github.com/openwisp/openwisp-radius/blob/602f3ecf357c3b4ecba0a60dabedd6f9c28f1319/openwisp_radius/api/utils.py#L55-L62

### PR DESCRIPTION
**Pull request by iamvaishnav**
closes #301

### CODE REVIEW

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
@iamvaishnav the build is faling for commit message guideline issues, please read that part of the docs more carefully, it's linked in the build failure.

Codewise looks good, I wonder if an additional test here may be useful or not, @pandafy if you could look into this as well it would be helpful, thanks.

File: openwisp_radius/base/models.py [hunk 1]
```diff
@@ -1139,6 +1139,11 @@ def allowed_mobile_prefixes_list(self):
             mobile_prefixes = self.allowed_mobile_prefixes.split(',')
         return mobile_prefixes
 
+    def get_sms_verification_enabled(self):
+        if self.sms_verification is None:
+            return app_settings.SMS_DEFAULT_VERIFICATION
+        return self.sms_verification
+
     def clean(self):
         if self.sms_verification and not self.sms_sender:
```
Comment (pandafy) [inline openwisp_radius/base/models.py:1148]: This needs to be updated to use `get_sms_verification()` method.

File: openwisp_radius/base/models.py [hunk 2]
```diff
@@ -1139,6 +1139,11 @@ def allowed_mobile_prefixes_list(self):
             mobile_prefixes = self.allowed_mobile_prefixes.split(',')
         return mobile_prefixes
 
+    def get_sms_verification_enabled(self):
```
Comment (pandafy) [inline openwisp_radius/base/models.py:1142]: Since the field name is `sms_verification`, it will be better to rename this to `get_sms_verification`.

**Review by pandafy — CHANGES_REQUESTED** [review]
I think a explicit unit test for `is_sms_verification_enabled` will be helpful to catch any bug that may appear in future.  

@nemesisdesign I found a discrepancy in the field definition for `sms_verification` and `registration_enabled`

https://github.com/openwisp/openwisp-radius/blob/04cb6754a35682f24d20de214605bea585a439c8/openwisp_radius/base/models.py#L1048-L1055

https://github.com/openwisp/openwisp-radius/blob/04cb6754a35682f24d20de214605bea585a439c8/openwisp_radius/base/models.py#L1116-L1118

Check `default` and `null` parameter in both of the fields. 

With current field definition, the work done by @iamvaishnav with never get executed, since the sms_verification field will never be `None`.

File: openwisp_radius/base/models.py (hunk 2 above)
Comment (iamvaishnav) [reply openwisp_radius/base/models.py:1142]: I will make the requested changes.

File: openwisp_radius/base/models.py (hunk 1 above)
Comment (iamvaishnav) [reply openwisp_radius/base/models.py:1148]: Are you suggesting to just return `self_verification` as it will never be `None`?

Comment (nemesisdesign) [reply openwisp_radius/base/models.py:1148]: I guess he meant `if self.get_sms_verification() and not self.sms_sender:`, right @pandafy?

Comment (pandafy) [reply openwisp_radius/base/models.py:1148]: Yes, you are right @nemesisdesign 
```suggestion
        if self.get_sms_verification() and not self.sms_sender:
```

**Comment by iamvaishnav** [comment]
> Check `default` and `null` parameter in both of the fields.

Should I add `null`  in `sms_verifications` field too? If so, how do I make migrations? Sorry if it sounds silly.

**Review by nemesisdesign — COMMENTED** [review]
I see a few things that are not very consistent with the rest of the codebase here which would be great to fix in this PR:

- I would rename `SMS_DEFAULT_VERIFICATION` to `SMS_VERIFICATION_ENABLED`
- I would allow the field to be `None` as the other fields, if the field is None it means the gobal setting should be used ( SMS_VERIFICATION_ENABLED`)
- I would document this setting in https://github.com/openwisp/openwisp-radius/blob/master/docs/source/user/settings.rst 

How does this sound @iamvaishnav @pandafy?

**Comment by iamvaishnav** [comment]
> I see a few things that are not very consistent with the rest of the codebase here which would be great to fix in this PR:
> 
> * I would rename `SMS_DEFAULT_VERIFICATION` to `SMS_VERIFICATION_ENABLED`
> * I would allow the field to be `None` as the other fields, if the field is None it means the gobal setting should be used (`SMS_VERIFICATION_ENABLED`)
> * I would document this setting in https://github.com/openwisp/openwisp-radius/blob/master/docs/source/user/settings.rst
> 
> How does this sound @iamvaishnav @pandafy?

Sounds good to me.

**Comment by nemesisdesign** [comment]
@iamvaishnav is this ready for review yet?

**Comment by iamvaishnav** [comment]
I don't know why it still fails.

**Comment by codesankalp** [comment]
> I don't know why it still fails.

It needs to be updated with the master branch.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Great progress @iamvaishnav, there's a few inconsistencies to adjust from a visual point of view.

Let's see:

Screenshot from 2021-11-03 19-08-56

- I think we should move the sms option after "Needs identity verification:".
- The allowed options of the select should be identical to the fields above, while now are very different.

I think it's because this kind of code is missing:

https://github.com/openwisp/openwisp-radius/blob/ee2d4f37ad3859aec08efc02d4ad3d8531863dc6/openwisp_radius/admin.py#L588-L603

I don't see any change in the admin.

**Comment by iamvaishnav** [comment]
> I don't see any change in admin.py, but I think it's necessary to do a bit more work to achieve consistency, can you do it please?

Will do.

**Comment by nemesisdesign** [comment]
> > I don't see any change in admin.py, but I think it's necessary to do a bit more work to achieve consistency, can you do it please?
> 
> Will do.

Cool, ping us when ready @iamvaishnav :+1:

**Comment by iamvaishnav** [comment]
@nemesisdesign Could you review it?

File: openwisp_radius/migrations/0025_sms_verification.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 3.1.13 on 2021-11-12 15:01
```
Comment (pandafy) [inline openwisp_radius/migrations/0025_sms_verification.py:1]: Why there are two migrations files? Can you merge them into one?

File: openwisp_radius/admin.py [hunk 4]
```diff
@@ -617,6 +618,22 @@ class OrganizationRadiusSettingsForm(AlwaysHasChangedMixin, forms.ModelForm):
         help_text=_IDENTITY_VERIFICATION_ENABLED_HELP_TEXT,
         fallback='',
```
Comment (pandafy) [inline openwisp_radius/admin.py:619]: @nemesisdesign shouldn't the fallback value be set to `False` here?

**Review by pandafy — CHANGES_REQUESTED** [review]
Good work @iamvaishnav, we are near completion of this PR. 
I requested a change below. After that it should be ready to merge. 

Can you also rebase on the current master branch?

File: openwisp_radius/migrations/0025_sms_verification.py (hunk 3 above)
Comment (iamvaishnav) [reply openwisp_radius/migrations/0025_sms_verification.py:1]: Will do.

File: openwisp_radius/admin.py (hunk 4 above)
Comment (nemesisdesign) [reply openwisp_radius/admin.py:623]: @pandafy why should it be?

Comment (pandafy) [reply openwisp_radius/admin.py:623]: You are right, this is correct.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
@iamvaishnav looks good! Please can you upgrade your version of black and run `openwisp-qa-format`? Then let's just make sure the build passes and we're done.

**Comment by iamvaishnav** [comment]
> Please can you upgrade your version of black and run `openwisp-qa-format`

`openwisp-qa-format` is missing in this repo.

**Comment by nemesisdesign** [comment]
> > Please can you upgrade your version of black and run `openwisp-qa-format`
> 
> `openwisp-qa-format` is missing in this repo.

Please read this section carefully: https://openwisp.io/docs/developer/contributing.html#python-code-conventions.
Install the latest master of openwisp-utils in your virtualenv.

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
The build is now failing because of the migration in sample_radius, please can you fix it?
Are you running `./run-qa-checks` locally before pushing as explained in https://openwisp.io/docs/developer/contributing.html#python-code-conventions?

**Comment by iamvaishnav** [comment]
> Are you running `./run-qa-checks` locally before pushing as explained in

Yes

**Review by nemesisdesign — COMMENTED** [review]
Build passes, @pandafy let's do one final manual check and then once all is confirmed to be good we can merge :+1:.

**Review by pandafy — APPROVED** [review]
I did a manual testing and everything works as expected. 

LGTM 👍🏼

**Review by nemesisdesign — APPROVED** [review]
Thanks @iamvaishnav @pandafy :+1:

**Pull request merged**